### PR TITLE
GH-1884

### LINKED ISSUE #1884: DicomDataset FunctionalGroupValues method crashed in case of an empty DicomTag.SharedFunctionalGroupsSequence
**Describe the bug**
The method `public DicomDataset DicomDataset.FunctionalGroupValues(int frame)` crashes with an `IndexOutOfBoundsException` in case an empty `SharedFunctionalGroupsSequence` is provided to the DICOM dataset.

**To Reproduce**
```
//Arrange
var dataset = new DicomDataset();
var sequence = new DicomSequence(DicomTag.SharedFunctionalGroupsSequence);
dataset.Add(sequence);

//Act
var result = dataset.FunctionalGroupValues(0); //Boom!
```

**Expected behavior**
If the sequence is empty, it shouldn't crash and preferably it just skips the shared functional group sequence part.

**Environment**
Fellow Oak DICOM version: 5.1.3 (latest)
OS: Win11 x64
Platform: .NET 8

### PR DESCRIPTION
**Pull request by pvrobays**
fix index out of bounds in FunctionalGroupValues in case of an empty SharedFunctionalGroupsSequence

Fixes #1884 .

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Don't crash in case of an empty SharedFunctionalGroupsSequence in DicomDataset.FunctionalGroupValues, just skip the sequence

### CODE REVIEW

File: FO-DICOM.Core/DicomDataset.cs [hunk 1]
```diff
@@ -806,16 +807,20 @@ public DicomDataset FunctionalGroupValues(int frame)
...
-                            {
-                                functionalDs.AddOrUpdate(item);
-                            }
+                            continue;
+                        }
+
+                        foreach (var item in sequence.Items[0])
+                        {
+                            functionalDs.AddOrUpdate(item);
                         }
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/DicomDataset.cs:818]: Is there a reason to change this?

File: FO-DICOM.Core/DicomDataset.cs [hunk 2]
```diff
@@ -829,17 +834,20 @@ public DicomDataset FunctionalGroupValues(int frame)
...
+                            continue;
+                        }
+
+                        foreach (var item in sequence.Items[0])
                         {
-                            foreach (var item in sequence.Items[0])
-                            {
-                                functionalDs.AddOrUpdate(item);
-                            }
+                            functionalDs.AddOrUpdate(item);
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/DicomDataset.cs:844]: Same here. I guess it's a matter of taste...

**Review by mrbean-bremen — COMMENTED** [review]
Thanks - can you also adapt the release notes please?

File: FO-DICOM.Core/DicomDataset.cs (hunk 2 above)
Comment (pvrobays) [reply FO-DICOM.Core/DicomDataset.cs:844]: Definitely a matter of taste. One of my plugins concerning code complexity just noticed the cognitive complexity of the method was high, and that seemed like an easy win.

Comment (mrbean-bremen) [reply FO-DICOM.Core/DicomDataset.cs:844]: Ok - I'm ok with that, thanks!

File: ChangeLog.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 - Fix bug where under some circumstances (transcode a 8 bit image with odd row or column length from explicit to implicit TS) a odd length pixeldata is written (#1403)
 - Fix bug where DicomDataset.GetDicomTag thew an exception if the private tag does not exist in dataset (#1840)
 - FO-DICOM.Tests target net8.0-windows instead of net7.0-windows
+- Fix issue where DicomDataset.FunctionalGroupValues throws IndexOutOfBounds in case of en empty SharedFunctionalGroupSequence (#1884)
```
Comment (mrbean-bremen) [inline ChangeLog.md]: typo: "en empty"

**Review by mrbean-bremen — APPROVED** [review]
Thanks!

**Review by gofal — APPROVED** [review]

**Pull request merged**